python3Packages.ezdxf: 1.3.2 -> 1.4.3, add updateScript, add withGui argument

mjoerg opened the pull request:
https://github.com/mozman/ezdxf/blob/refs/tags/v1.4.3/notes/pages/CHANGELOG.md
https://github.com/mozman/ezdxf/compare/refs/tags/v1.3.2...refs/tags/v1.4.3

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`
Commit: `493f6d1f1c47bb17af70cc78205fb950f00c07e6`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>meerk40t</li>
    <li>meerk40t.dist</li>
    <li>python312Packages.ezdxf</li>
    <li>python312Packages.ezdxf.dist</li>
    <li>python313Packages.ezdxf</li>
    <li>python313Packages.ezdxf.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>meerk40t</li>
    <li>meerk40t.dist</li>
    <li>python312Packages.ezdxf</li>
    <li>python312Packages.ezdxf.dist</li>
    <li>python313Packages.ezdxf</li>
    <li>python313Packages.ezdxf.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>meerk40t</li>
    <li>meerk40t.dist</li>
    <li>python312Packages.ezdxf</li>
    <li>python312Packages.ezdxf.dist</li>
    <li>python313Packages.ezdxf</li>
    <li>python313Packages.ezdxf.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>meerk40t</li>
    <li>meerk40t.dist</li>
    <li>python312Packages.ezdxf</li>
    <li>python312Packages.ezdxf.dist</li>
    <li>python313Packages.ezdxf</li>
    <li>python313Packages.ezdxf.dist</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5442

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-in-distress/3604/122

doronbehar reviewed (commented):
I'd also squash the 1.4.0 -> 1.4.3 commits.

doronbehar commented on pkgs/development/python-modules/ezdxf/default.nix:
```diff
@@ -87,6 +88,11 @@ buildPythonPackage rec {
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+    allowedVersions = "^[0-9]+\\.[0-9]+\\.[0-9]+$";
+  };
```
> Do you mean `nixpkgs-update` instead of `nixpkgs-review`?

Yes, sorry :)

> I think the versions should still be filtered, as otherwise, unwanted versions are still picked up ([#442194 (comment)](https://github.com/NixOS/nixpkgs/pull/442194#issuecomment-3324902869)).

I see. Thanks for the context. It'd be nice to put this detail in a comment above the `updateScript`.

mjoerg committed:
python3Packages.ezdxf: 1.3.2 -> 1.4.0

https://github.com/mozman/ezdxf/blob/refs/tags/v1.4.0/notes/pages/CHANGELOG.md
https://github.com/mozman/ezdxf/compare/refs/tags/v1.3.2...refs/tags/v1.4.0

mjoerg committed:
python3Packages.ezdxf: cleanup

mjoerg committed:
python3Packages.ezdxf: enable more tests

mjoerg committed:
python3Packages.ezdxf: add withGui argument

mjoerg committed:
python3Packages.ezdxf: add updateScript

mjoerg committed:
python3Packages.ezdxf: 1.4.0 -> 1.4.3

https://github.com/mozman/ezdxf/blob/refs/tags/v1.4.3/notes/pages/CHANGELOG.md
https://github.com/mozman/ezdxf/compare/refs/tags/v1.4.0...refs/tags/v1.4.3
